### PR TITLE
Add owning-a-home/explore-rates scripts and dependencies

### PR DESCRIPTION
**Pull request by anselmbradford**
Apps (such as owning-a-home) may have their own esoteric dependencies that are not applicable to other aspects of consumerfinance.gov. It therefore makes sense to provide these projects with their own package.json to manage their own dependencies, which can be pulled into the webpack pipeline for that app's scripts.

## Additions

- Moves /explore-rates/ owning-a-home files into the project and adds dependencies.

## Changes

- Adds rate-checker dependencies to oah's package.json.

## Testing

1. Go into `urls.py` and change:
```
url(r'^owning-a-home/explore-rates/',
        include(oah.urls_for_prefix('explore-rates'))),
```

To:

```
url(r'^owning-a-home/explore-rates/',
        TemplateView.as_view(
        template_name='owning-a-home/explore-rates/index.html'),
        name='explore-rates'),
```

2. Restart the server.
3. Run `gulp clean`.
4. From project root, run `cp ./cfgov/unprocessed/apps/owning-a-home/example-config.json ./cfgov/unprocessed/apps/owning-a-home/config.json`.
5. Run `gulp scripts:apps`.
6. Verify message appears: `App dependencies not installed, please run from project root: npm --prefix ./cfgov/unprocessed/apps/owning-a-home install ./cfgov/unprocessed/apps/owning-a-home`.
7. Run  `npm --prefix ./cfgov/unprocessed/apps/owning-a-home install ./cfgov/unprocessed/apps/owning-a-home` from project root.
8. Re-run `gulp scripts:apps` and verify it passes.
9. Run `gulp build`.
10. Check that http://localhost:8000/owning-a-home/explore-rates/ has no console error. (Note: layout will not be correct, but this PR is just about getting the JS assets into that page without errors.)

## Notes
- Contains https://github.com/cfpb/cfgov-refresh/pull/3817
- Contains https://github.com/cfpb/cfgov-refresh/pull/3819
- Contains https://github.com/cfpb/cfgov-refresh/pull/3820

### CODE REVIEW

**Comment by sebworks** [comment]
@anselmbradford, the error you are getting is related to the mocha / imports statments. Fix is coming.

File: cfgov/cfgov/urls.py [hunk 1]
```diff
@@ -71,8 +71,16 @@ def flagged_wagtail_template_view(flag_name, template_name):
 
     url(r'^owning-a-home/closing-disclosure/',
         include(oah.urls_for_prefix('closing-disclosure'))),
+
     url(r'^owning-a-home/explore-rates/',
         include(oah.urls_for_prefix('explore-rates'))),
+
+    # TODO: uncomment when explore-rates CSS/JS is ready and tested.
```
Comment (sebworks) [inline cfgov/cfgov/urls.py]: Just flag this route.

**Comment by sebworks** [comment]
After you flag the route, then it's good to merge for me.

**Comment by anselmbradford** [comment]
Rebased latest changes in, please remember to delete and re-pull locally.

**Pull request merged**